File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {

    public String reverseVowels(String s) {
        Set<Character> vow = new HashSet<Character>(Arrays.asList('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'));
        char[] res = s.toCharArray();
        char ch;

        int l = 0;
        int r = s.length() - 1;
        while (l < r) {
            if (vow.contains(res[l]) && vow.contains(res[r])) {
                ch = res[l];
                res[l] = res[r];
                res[r] = ch;
                l++;
                r--;
            }
            if (vow.contains(res[l]) == false) {
                l++;
            }
            if (vow.contains(res[r]) == false) {
                r--;
            }
        }
        String ans = new String(res);
        return ans;
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vow = {'a','e','i','o','u', 'A','E','I','O','U'}
        res = list(s) 
        
        l = 0
        r = len(s) - 1                       
        while(l < r):
            if res[l] in vow and res[r] in vow:
                res[l], res[r] = res[r], res[l]
                l += 1
                r -= 1
            if res[l] not in vow:
                l += 1
            if res[r] not in vow:
                r -= 1
                
        return ''.join(res)
        